### PR TITLE
Add IPv6 support

### PR DESCRIPTION
**Pull request by paramite**
This patch makes bridge accept IPv6 addresses as amqp_url value.

### CODE REVIEW

**Review by csibbitt — APPROVED** [review]
This LGTM but I'd like to know it's tested before we merge.

**Comment by vkmc** [comment]
recheck

**Comment by vkmc** [comment]
There is some recurring failure in the CI, not related to the timeouts. It seems that the bridge container is constantly failing. See for example https://review.rdoproject.org/zuul/build/e7227852b3a747bf857a1168dddc1c71/log/controller/controller/post_oc_describe_pod_default-cloud1-ceil-event-smartgateway-7b5d4fcd79-5fswr.log in the [stf-crc-ocp_414-local_build](https://review.rdoproject.org/zuul/build/e7227852b3a747bf857a1168dddc1c71). We don't really have access to the container logs IIRC.

**Review by vkmc — CHANGES_REQUESTED** [review]
Seems there is something wrong when trying to start the smart-gateway with this change.

File: tests.c [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 #include "bridge.h"
 #include "minunit.h"
+#include "utils.h"
```
Comment (csibbitt) [inline tests.c:19]: Without this I get the following compile problem:

```
gcc -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -c -o test_obj/tests.o tests.c
tests.c: In function ‘test_match_amqp_url_hostname’:
tests.c:26:5: error: implicit declaration of function ‘match_regex’ [-Wimplicit-function-declaration]
   26 |     match_regex(AMQP_URL_REGEX, matches, 10,
      |     ^~~~~~~~~~~
make: *** [Makefile:94: test_obj/tests.o] Error 1
```

**Comment by vyzigold** [comment]
The ending of the new regex is the cause of the CI failures. It seems the "[^\\s]" doesn't work as intended instead of matching "non-whitespace" it seems to match "non-'s' character".  And because we use "s" in the address in the CI, it fails. Otherwise an impressive work. Thank you.

File: bridge.c [hunk 2]
```diff
@@ -275,16 +236,21 @@ int main(int argc, char **argv) {
 
     match_regex(AMQP_URL_REGEX, matches, 10, app.amqp_con.url);
     if (matches[3] != NULL) {
-        app.amqp_con.user = strdup(matches[2]);
+        app.amqp_con.user = strdup(matches[3]);
```
Comment (csibbitt) [inline bridge.c:239]: Good find! This is an old bug that we should have caught in the initial review years ago.

**Comment by csibbitt** [comment]
> The ending of the new regex is the cause of the CI failures. It seems the "[^\\s]" doesn't work as intended instead of matching "non-whitespace" it seems to match "non-'s' character". And because we use "s" in the address in the CI, it fails. Otherwise an impressive work. Thank you.

I reproduced this locally, if I change /baz to /bas in a unit test I get a segfault instead of a pass. :)

**Review by csibbitt — APPROVED** [review]

File: tests.c (hunk 1 above)
Comment (paramite) [reply tests.c:19]: Yes you need utils as I moved match_regex there to be able to include it both in bridge.c and tests.c and not conflict with two main functions.

**Comment by paramite** [comment]
recheck

**Comment by paramite** [comment]
recheck

**Comment by vkmc** [comment]
recheck

**Review by vkmc — APPROVED** [review]
Code looks good, and thanks so much for enhancing test coverage and automation as well!

**Pull request merged**